apps/nextjs/src/utils/trpc.ts

```diff
@@ -2,12 +2,13 @@ import { type AppRouter } from "@agreeto/api";
 import { transformer } from "@agreeto/api/transformer";
 import { httpBatchLink } from "@trpc/client";
 import { createTRPCNext } from "@trpc/next";
+import { env } from "../env/server.mjs";
 
 const getBaseUrl = () => {
   if (typeof window !== "undefined") return ""; // browser should use relative url
-  if (process.env.VERCEL_URL) return `https://${process.env.VERCEL_URL}`; // SSR should use vercel url
+  if (env.VERCEL_URL) return `https://${env.VERCEL_URL}`; // SSR should use vercel url
 
-  return `http://localhost:${process.env.PORT ?? 3000}`; // dev SSR should use localhost
+  return `http://${env.HOST}:${env.PORT ?? 3000}`; // dev SSR should use localhost
```

Comment:
@juliusmarminge and reused the server env here as well.

Reply:
this won't work, since this code is executed on the client, either rename the HOST to NEXT_PUBLIC_HOST or just hardcode localhost

i also think PORT is poly filled and we don't need to explicitly state it

Reply:
The client execution shouldn't even reach line 11 but return on [L8](https://github.com/agreeto/agreeto/pull/6/commits/23b90b95d6fa1ebf14266c82b21ce7bebb6c527a#diff-54718864100eec4fe1b5dde95f05214582187c00d019caee755a74273fa5030fR8) if I understand this correctly.

The comment mentions SSR, so I thought it's OK to reference server env vars?

Reply:
Yes problem is not referencing server vars, its the import statement 

When importing from "env/server" we trigger validation which will throw an error